cardano-db-sync/src/Cardano/DbSync/Ledger/Types.hs

```diff
@@ -193,3 +198,97 @@ instance Anchorable (WithOrigin SlotNo) CardanoLedgerState CardanoLedgerState wh
   getAnchorMeasure _ = getTipSlot . clsState
 
 data SnapshotPoint = OnDisk LedgerStateFile | InMemory CardanoPoint
+
+-- | Per-era pure getters and setters on @NewEpochState@. Note this is a bit of an abuse
+-- of the cardano-ledger/ouroboros-consensus public APIs, because ledger state is not
+-- designed to be updated this way. We are only replaying the chain, so this should be
+-- safe.
+class HasNewEpochState era where
```

Comment:
That's a very useful class to have, which can be reused for https://github.com/orgs/IntersectMBO/projects/8/views/3?pane=issue&itemId=60533809